packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -63,9 +63,12 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
             isEnabled: () => this.workspaceService.opened,
             execute: () => this.closeWorkspace()
         });
-        commands.registerCommand(WorkspaceCommands.OPEN_RECENT_WORKSPACE, {
-            isEnabled: () => this.workspaceService.hasHistory,
-            execute: () => this.quickOpenWorkspace.select()
+        this.workspaceService.recentWorkspaces().then(({ length }) => {
+            if (!length) {
```

Comment:
Don't we need the opposite logic here? 
I may be wrong but since we are waiting for the promise to resolve before registering the command, we are never able to create the menu item (our `File ⮕ Open Recent Workspace` menu never shows up anymore) since it relies on it.